### PR TITLE
[react-table]: fix types for TableSortByToggleProps & TableGroupByToggleProps

### PR DESCRIPTION
**Pull request by ggascoigne**
This is based on https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57539, but expanding that change to TableGroupByToggleProps, and adding tests for both of these changes.

I'd have just suggested changes to the other PR, but it's a pest to suggest changes in a file that wasn't in the PR and contrary to the description, that PR didn't update or add tests.

unrelated, this PR updated the tsconfig.json to match the current pattern.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [useSortBy](https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useSortBy.js#L33-L54) & [useGroupBy](https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useGroupBy.js#L37-L51)

### CODE REVIEW

**Comment by ggascoigne** [comment]
Grr - to run the tests locally the tsconfig needs one thing, but to run in CI it needs another.  Reverting that extraneous change then :(

File: types/react-table/index.d.ts
Comment (amcasey) [inline types/react-table/index.d.ts:804]: Can you please confirm that this is not supposed to be optional (e.g. as on `TableCommonProps`)?

Comment (amcasey) [reply types/react-table/index.d.ts:804]: Same for `onClick`.

**Comment by amcasey** [comment]
FYI @jdmg94

**Review by amcasey — CHANGES_REQUESTED** [review]
Quick question

File: types/react-table/index.d.ts
Comment (ggascoigne) [reply types/react-table/index.d.ts:804]: see:

https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useSortBy.js#L33-L54
https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useGroupBy.js#L37-L51

That said, since none of these values are documented anywhere (so might change) I think that I'm going to make them all be optional since I think that that reflects the contract (or lack thereof) better.

Comment (amcasey) [reply types/react-table/index.d.ts]: Thanks!

**Review by amcasey — APPROVED** [review]
LGTM.  I'll leave it up to you to decide whether you want to wait for sign-offs from other owners.

**Comment by ggascoigne** [comment]
Ready to merge

**Comment by dkozickis** [comment]
CleanShot 2021-12-10 at 10 01 40@2x

**Comment by amcasey** [comment]
Thanks, @dkozickis!

@ggascoigne can you please add a test corresponding to the doc example and update the typings appropriately?

**Comment by ggascoigne** [comment]
@dkozickis do you have a link handy to the example that causing problems?

**Pull request merged**
